clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -104,6 +132,45 @@ public long requestLatencyMs() {
         return latencyMs;
     }
 
+    private void releaseBuffer() {
+        if (memoryPool != null && responsePayload != null) {
```

Comment:
Feels like this should be calling usingMemoryPool()

Reply:
No it should be. MemoryPool implementation shouldn't be dependent here whether we return the buffer to memorypool or not.